app/src/androidTest/java/org/kiwix/kiwixmobile/settings/SettingsRobot.kt

```diff
@@ -45,10 +43,8 @@ fun settingsRobo(func: SettingsRobot.() -> Unit) =
 class SettingsRobot : BaseRobot() {
 
   private fun clickRecyclerViewItems(@StringRes vararg stringIds: Int) {
-    val matchers: Array<Matcher<View>?> = arrayOfNulls(stringIds.size)
-    for (i in stringIds.indices) {
-      matchers[i] = withText(stringIds[i])
-    }
+    val matchers = stringIds.indices.map { withText(stringIds[it]) }.toTypedArray()
```

Comment:
you don't need to use the indices, use map directly.

Reply:
>How about going without a variable declaration? I tried this but it looked quite overwhelming to see, thoughts?

The variable is probably not needed, sometimes a private extension function can say things quite clearly if a line is overlong

Reply:
How can we map stringIds to indices of map without even using indices? Sounds quite weird, but let me see

Reply:
`arrayOf(1,2,3).map { it.toString() } == listOf("1","2","3")`

a map function operates on the items of an iterable returning a new list containing the results of the transformation operation specified by `map`

Reply:
I tried `stringIds.map { withText(stringIds[it]) }.toTypedArray()` just after seeing your very first comment today but this didn't work.
What's wrong in it, it indeed mapped each index of stringId with its corresponding _string_

Reply:
lets write this out explicitly

```
val aListOfInts:List<Int> = listOf(1,2,3)
val doubledList:List<Int> = aListOfInts.map { anInt:Int -> anInt * anInt }
doubledList == listOf(1,4,9)
```

does that all make sense?

Reply:
I feel, i understood what you said(_a map function operates on the items of an iterable returning a new list containing the results of the transformation operation specified by `map`_), now building upon the same concept let's explain the following in
[test-commit](https://github.com/kiwix/kiwix-android/pull/2328/commits/ba37aacfdd8fb394b747001fbbd2df681b879dbd)  

* `val matchers1 = stringIds.map { withText(context.getString(stringIds[it])) }.toTypedArray()`

So what it does is iterates through all the stringIds and transform them to their corresponding strings, then why is it still failing the test???

What's wrong here? whereas using same passed the tests previously when _indices_ were being used and they sounded much more logical, clear and easier to grasp with, i'm in no sense saying to stick with the older version but asking where i'm going wrong now(in the one to be implemented without indices).

PS: No snippets please, you just give me code so I don't have to think much on it, so please don't do this until i feel helpless and explicitly ask the same.

Reply:
Look at the original java

```
    Matcher<View>[] matchers = new Matcher[stringIds.length];
    for (int i = 0; i < stringIds.length; i++) {
      matchers[i] = withText(stringIds[i]);
    }
```

And perhaps look at the `map` [docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map.html), the method signature is very illuminating

Reply:
additionally have I told you about [scratches](https://www.jetbrains.com/help/idea/scratches.html)? Or the[ Kotlin REPL](https://kotlinlang.org/docs/tutorials/quick-run.html) in android studio? Or even [kotlin playground](https://play.kotlinlang.org/)? There is a lot of ways to write some kotlin and run it instantly if you don't feel 100% confident in an api and want to experiment

Reply:
Yeah, you did tell me about the scratches and playground, i don't use playground as it was never handy to me as i always had to create a main function. And i knew about the existence of scratches but didn't know how to use 'em, now i learned using them and it helped :) Scratches with REPL work awesome!
Thanks!